src/k8s/pkg/k8s/cluster/cluster.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/canonical/microcluster/client"
+	"github.com/canonical/microcluster/microcluster"
+)
+
+// Bootstrap sets up new cluster and returns the information about the daemon.
+func Bootstrap(ctx context.Context, opts ClusterOpts) (ClusterMember, error) {
+	m, err := microcluster.App(ctx, microcluster.Args{
+		Debug:      opts.Debug,
+		ListenPort: opts.Port,
+		StateDir:   opts.StorageDir,
+		Verbose:    opts.Verbose,
+	})
+	if err != nil {
+		return ClusterMember{}, fmt.Errorf("failed to configure cluster: %w", err)
+	}
+
+	// Get system hostname.
+	hostname, err := os.Hostname()
+	if err != nil {
+		return ClusterMember{}, fmt.Errorf("failed to retrieve system hostname: %w", err)
+	}
+
+	port, err := strconv.Atoi(opts.Port)
+	if err != nil {
+		return ClusterMember{}, fmt.Errorf("failed to parse Port: %w", err)
+	}
+	// Get system address.
+	address := util.CanonicalNetworkAddress(
+		util.NetworkInterfaceAddress(), port,
+	)
+
+	member := ClusterMember{
+		Name:    hostname,
+		Address: address,
+	}
+	err = m.NewCluster(hostname, address, time.Second*30)
+	return member, err
+}
+
+// NewClient returns a client to interact with the cluster.
+// It will return:
+//   - a local client, if executing node is part of cluster (valid config.stateDir)
+//   - a remote client, if executing node is part of cluster but config.Address is set
+//
+// TODO(bschimke):
+//   - a REST client, if executed from outside of a cluster. A REST client has limited functionality.
+//     This requires a mechanism to distribute the server certificates to the client
+func NewClient(ctx context.Context, opts ClusterOpts) (*Client, error) {
+	err := opts.isValid()
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	m, err := microcluster.App(ctx, microcluster.Args{
+		Debug:      opts.Debug,
+		ListenPort: opts.Port,
+		StateDir:   opts.StorageDir,
+		Verbose:    opts.Verbose,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cannot read cluster info: %w", err)
+	}
+
+	var microClient *client.Client
+	if opts.Address != "" {
+		microClient, err = m.RemoteClient(opts.Address)
+	} else {
+		microClient, err = m.LocalClient()
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %w", err)
+	}
+
+	return &Client{
+		microClient: microClient,
+	}, nil
+}
+
+// Client is a wrapper around the MicroCluster client
+type Client struct {
+	microClient *client.Client
+}
+
+// GetMembers returns information about all members of the cluster.
+func (c *Client) GetMembers(ctx context.Context) ([]ClusterMember, error) {
+	clusterMembers, err := c.microClient.GetClusterMembers(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	members := make([]ClusterMember, len(clusterMembers))
+	for i, clusterMember := range clusterMembers {
+		fingerprint, err := shared.CertFingerprintStr(clusterMember.Certificate.String())
+		if err != nil {
+			continue
+		}
+
+		members[i] = ClusterMember{
+			clusterMember.Name,
```

Comment:
Use named fields instead, also for other fields here.
```suggestion
			Name: clusterMember.Name,
```

Reply:
Done